### ISSUE TITLE
Lighten the color of Placeholder text in Basis Theme

### ISSUE THREAD

**Issue by philsward** [backdrop/backdrop-issues#5249]
**Description of the need**

The color for placeholder text is too close in color to the theme text, making it difficult to understand "There's nothing actually in this box".

The UX is confusing when looking at it and confusing when clicking on it (because placeholder text doesn't disappear until you start typing).

This could potentially result in people clicking the input form trying to delete the text when there isn't anything to actually delete.

**Proposed solution**

Lightening the color will provide contrast between the theme text and placeholder text, which in theory will aid with indicators that the text inside the input form isn't actually text that must be deleted.

The proposed color is something closer to HEX: #999

**Alternatives that have been considered**

Not using placeholders...

**Additional information**

Examples:

**Current:**
![image](https://user-images.githubusercontent.com/3436110/134289455-dccd632e-97c5-4056-a0d5-8f9838addd87.png)

**Proposed:**
![image](https://user-images.githubusercontent.com/3436110/134289493-6af322b3-d70e-4906-9001-24b2fcba52fc.png)

**Comment by indigoxela** [comment]
Please also consider web accessibility. From your screenshot I'd assume that the contrast is too low.

**Comment by olafgrabienski** [comment]
> Please also consider web accessibility.

I guess the current used color for placeholders in *Basis* is around `#757575`, which has a contrast ratio of `4.6:1`. According to [MDN Web docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast) the minimum ratio (AA rating) is `4.5:1`.

Seems we can't lighten the text without creating an accessability issue. I'm wondering if there's any approach which meets *both*:

- understandable contrast between different text states
- enough contrast between background and text for legibility

**Comment by philsward** [comment]
@olafgrabienski is that contrast ratio specific to placeholders or just a specific ratio recommend for general site viewing?

The point of placeholders yes is to give an example, but more importantly, it needs to be shown that it is ghost text, or text that isn't really there.

Maybe a different color?

I didn't play with it (just now thought about it), but one idea is to blank the placeholder when the input form has focus. This would honestly be a better UX all the way around.

**Comment by olafgrabienski** [comment]
@philsward It's a recommendation for "readable interfaces" in general.

**Comment by klonos** [comment]
@philsward Drupal/Backdrop core does not offer a way to add placeholder text to fields OOTB. You either need to port and use a contrib module, or add the placeholder attribute via a custom module/theme (that perhaps converts the help text into a placeholder). How are you achieving that?

> The UX is confusing when looking at it and confusing when clicking on it (because placeholder text doesn't disappear until you start typing).

I see how that may be an issue, but there are solutions available, that can be implemented on the theme level. Seehttps://stackoverflow.com/questions/9707021/how-do-i-auto-hide-placeholder-text-upon-focus-using-css-or-jquery for example, where this is being suggested in order to hide the placeholder text on focus instead of when you start typing:

```CSS
input:focus::placeholder {
  color: transparent;
}
```

**Comment by philsward** [comment]
@klonos I agree this can be fixed at the theme level, I just don't want to have to fix it for every site and override that theme for every site where it is used, then have to maintain the code base every time basis is updated.

This isn't a personal preference tweak, this is a poor UX tweak.

**Comment by philsward** [comment]
List of other selectors for various browsers:

https://stackoverflow.com/questions/13183421/how-to-change-placeholder-color-on-focus

```
input:focus::placeholder {
  color: transparent;
}
```

This worked great on a Seven theme override I needed and it gives the UX a feel that is lot more natural. Hope to see this gain some traction for the Basis theme so modules like webform which use placeholders can benefit from it.

**Comment by klonos** [comment]
> Backdrop core does not offer a way to add placeholder text to fields OOTB....

@jenlampton pointed out that the search field on our search block can be configured to have the a placeholder text:

<img width=50% src="https://user-images.githubusercontent.com/2423362/135666068-985e5e97-dd80-49bf-875b-a4f4ce9c76f6.png">

Another place where we use placeholder text (although it may not be subject to theme colors) is the search field in the admin bar.

**Comment by docwilmot** [comment]
Also, Backdrop allows placeholders via FormAPI: https://docs.backdropcms.org/form_api#placeholder

**Comment by philsward** [comment]
Should I rename the title to reflect blanking the placeholder? I'm still a fan of making the color slightly lighter (not much) but I think the better UX is to blank the form field on focus.

**Comment by klonos** [comment]
@philsward I think that you have better chances with the blank-on-focus approach, unless you can specify a color that at the same time satisfies both of these conditions:
1. is "light enough" (to denote that it is not the actual text, but rather a placeholder)
2. is at least WCAG 2 level AA conformant (you can use https://webaim.org/resources/contrastchecker for example). If we choose anything less than that just because it is aesthetically pleasing, then I'm afraid that someone will at some point raise an accessibility issue, and that color will be changed again.
